### DISCUSSION TITLE
Is mice.impute.polr a "proper" imputation approach?

### DISCUSSION THREAD

**Discussion by nmorris-lilly** [amices/mice#358] [Missing data methodology]
Looking at the 'mice.impute.polr' code (and documentation), it appears that the common step used in other imputation algorithms "Draw BETA from N(bhat, V(bhat))" is not performed. Just wondering if I understood correctly, and if their is some reason for skipping this step. Does this make the imputation not "proper?" Would it be possible to include this extras step in the function? 

Apologies if I have misunderstood something as I am still new to MI approaches.

**Comment by stefvanbuuren** [comment]
Yes, your understanding is correct. The procedure is not Rubin-proper since it fails to draw parameters. I wrote this routine more than 20 years ago, and at that time it wasn't obvious (to me at least) how to draw the beta's for `polr` and `multinom`. 

My feeling is that non-properness is not a big issue for these models. Some limited simulations in [Table 3 of the FCS 2006 paper](
https://stefvanbuuren.name/publications/FCS%20in%20multivariate%20imputation%20-%20JSCS%202006.pdf) showed that the coverage of the confidence interval of the procedure is almost perfect. If there is a problem, then it is likely to occur for sample sizes substantially smaller than 412.

We may follow one of two routes to make it proper:
- *Bayesian*: Nowadays, we could probably borrow from `stan` or the `mi` package for procedures that draw parameters, although I am not sure whether there would be a Bayesian equivalent for `multinom()`;
- *Bootstrap*: Draw a bootstrap sample before fitting the model, and use the estimated beta's. This is easy to program. See how it is done in the `mice.impute.norm.boot()` function.

I have no inclination to do any more work on this, but if anyone feels up to it, I'd be interested in any simulations that would illustrate whether being proper actually matters for these models.
